### DISCUSSION TITLE
How can we use YoutTubePlayerScaffold with a deeply nested structure?

### DISCUSSION THREAD

**Discussion by mark8044** [sarbagyastha/youtube_player_flutter#688] [Q&A]
Right now I have an outer `Scaffold` with multiple levels of deeply nested widgets.  Inside one of those deeply nested widgets (about 5-6 levels deep) is where my `YoutubePlayer` widget sits.  My `YoutubePlayer` widget is a stateful widget which dynamically gets its `videoId` based on RESTful data pulled from a remote source.

With this new `YoutTubePlayerScaffold` method, I have to place it above the main `Scaffold` and define the controller very early on.  Then I have to use `builder` to pass a `player` param down 5-6 levels of nesting.  Then within the nested level I have to pass a Youtube video ID back up to the root, probably via some kind of state management system.

This seems very very tedious.  Is it possible to have the YoutubePlayerScaffold exist without the controller which can at least be controlled at the `YoutubePlayer` widget level?

This part seemed to be working much better pre 3.x

**Comment by sarbagyastha** [comment]
```dart 

final controller = YoutubePlayerController();

YoutubePlayerScaffold(
  controller: controller,
  builder: (context, player) {
    return NestedWidget();
  }
),


NestedWidget{ 
   initState() {
      fetchIdFromRest.then((id) {
         context.ytController.loadVideoById(videoId: id);
      });
   }

   build(context) {
      return Scaffold(
        body: YoutubePlayer.of(context), // I'll add this in the next update.
      );
   }
}

```

@mark8044 Can you not do something like this ?

**Reply by mark8044** [reply]
Seems interesting, I can try that when `YoutubePlayer.of(context)` is working
